src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -498,6 +498,7 @@ class DiffusionPipeline(ConfigMixin, PushToHubMixin):
     _exclude_from_cpu_offload = []
     _load_connected_pipes = False
     _is_onnx = False
+    _offload_gpu_id = None
```

Comment:
is this necessary? `_offload_gpu_id` should not be a class variable IMO

Reply:
This line would throw an error if it isn't defined anywhere. 
https://github.com/huggingface/diffusers/blob/514d2501c3f2d361a5fbcccbb60e5e88b381efa9/src/diffusers/pipelines/pipeline_utils.py#L1364

I added it as a class variable based on this comment

```
# _offload_gpu_id should be set to passed gpu_id (or id in passed `device`) or default to previously set id or default to 0
```
Assumed that `previously set id`  meant that the user could set it before hand.

Reply:
Ah yeah you're 100% right. Could we instead though do the following:

```py
self._offload_gpu_id = gpu_id or torch_device.index or getattr(self, "_offload_gpu_id", 0)
```

both here https://github.com/huggingface/diffusers/blob/514d2501c3f2d361a5fbcccbb60e5e88b381efa9/src/diffusers/pipelines/pipeline_utils.py#L1364 and here: https://github.com/huggingface/diffusers/blob/514d2501c3f2d361a5fbcccbb60e5e88b381efa9/src/diffusers/pipelines/pipeline_utils.py#L1449